plugin/core/sessions.py

```diff
@@ -148,15 +241,22 @@ def get_capability(self, capability: str) -> Optional[Any]:
     def should_notify_did_open(self) -> bool:
         textsync = self.capabilities.get('textDocumentSync')
         if isinstance(textsync, dict):
-            return bool(textsync.get('openClose'))
+            open_close = textsync.get('openClose')
+            if isinstance(open_close, dict):
+                return True  # dynamic registration
+            return bool(open_close)
         if isinstance(textsync, int):
             return textsync > TextDocumentSyncKindNone
         return False
 
     def text_sync_kind(self) -> int:
         textsync = self.capabilities.get('textDocumentSync')
         if isinstance(textsync, dict):
-            return int(textsync.get('change', TextDocumentSyncKindNone))
+            change = textsync.get('change', TextDocumentSyncKindNone)
+            if isinstance(change, dict):
+                # dynamic registration
+                return TextDocumentSyncKindIncremental  # or TextDocumentSyncKindFull?
```

Comment:
I wonder about that...

Reply:
I would like to test this in the st4000-exploration branch with both jedi and omnisharp and see what happens.

Reply:
If it turns out that a dynamic registration of `textDocument/didChange` notifications only means "full sync", then we'll just put full sync here.